freqtrade/plugins/pairlist/VolatilityFilter.py

```diff
@@ -0,0 +1,120 @@
+
+        self._days = pairlistconfig.get('lookback_days', 10)
+        self._min_volatility = pairlistconfig.get('min_volatility', 0)
+        self._max_volatility = pairlistconfig.get('max_volatility', sys.maxsize)
+        self._refresh_period = pairlistconfig.get('refresh_period', 1440)
+
+        self._pair_cache: TTLCache = TTLCache(maxsize=1000, ttl=self._refresh_period)
+
+        if self._days < 1:
+            raise OperationalException("VolatilityFilter requires lookback_days to be >= 1")
+        if self._days > exchange.ohlcv_candle_limit('1d'):
+            raise OperationalException("VolatilityFilter requires lookback_days to not "
+                                       "exceed exchange max request size "
+                                       f"({exchange.ohlcv_candle_limit('1d')})")
+
+    @property
+    def needstickers(self) -> bool:
+        """
+        Boolean property defining if tickers are necessary.
+        If no Pairlist requires tickers, an empty List is passed
+        as tickers argument to filter_pairlist
+        """
+        return False
+
+    def short_desc(self) -> str:
+        """
+        Short whitelist method description - used for startup-messages
+        """
+        return (f"{self.name} - Filtering pairs with volatility range "
+                f"{self._min_volatility}-{self._max_volatility} "
+                f" the last {self._days} {plural(self._days, 'day')}.")
+
+    def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Validate trading range
+        :param pairlist: pairlist to filter or sort
+        :param tickers: Tickers (from exchange.get_tickers()). May be cached.
+        :return: new allowlist
+        """
+        needed_pairs = [(p, '1d') for p in pairlist if p not in self._pair_cache]
+
+        since_ms = int(arrow.utcnow()
+                       .floor('day')
+                       .shift(days=-self._days - 1)
+                       .float_timestamp) * 1000
+        # Get all candles
+        candles = {}
+        if needed_pairs:
+            candles = self._exchange.refresh_latest_ohlcv(needed_pairs, since_ms=since_ms,
+                                                          cache=False)
+
+        if self._enabled:
+            for p in deepcopy(pairlist):
+                daily_candles = candles[(p, '1d')] if (p, '1d') in candles else None
+                if not self._validate_pair_loc(p, daily_candles):
+                    pairlist.remove(p)
+        return pairlist
+
+    def _validate_pair_loc(self, pair: str, daily_candles: Optional[DataFrame]) -> bool:
+        """
+        Validate trading range
+        :param pair: Pair that's currently validated
+        :param ticker: ticker dict as returned from ccxt.load_markets()
+        :return: True if the pair can stay, false if it should be removed
+        """
+        # Check symbol in cache
+        if pair in self._pair_cache:
+            return self._pair_cache[pair]
```

Comment:
```suggestion
        cached_res = self._pair_cache.get(pair, None)
        if cached_res is not None:
            return cached_res
```

Please change this section to the above code. 
otherwise we'll have a (rare) timing issue where the cache times out between the if and the next line.
This was recently changed (in 0f0607baec ) for RangeStabilityFilter too.